src/app/services/table.service.ts

```diff
@@ -1,39 +1,173 @@
 import {Injectable} from '@angular/core';
 import {Dimension} from '../models/table-models/dimension';
+import {DataTable} from '../models/table-models/data-table';
+import {Row} from '../models/table-models/row';
+import index from '@angular/cli/lib/cli';
 
 @Injectable()
 export class TableService {
 
-  private _rows: Dimension[];
-  private _columns: Dimension[];
+  private _dataTable: DataTable;
 
   constructor() {
-    this.rows = [];
-    this.columns = [];
+    this.dataTable = new DataTable();
+    this.mockData();
+  }
+
+  mockData() {
```

Comment:
the mocked data is only used for now to quickly create primeng table, later when there is backend connection, this function will be moved to unit test.